### PR TITLE
Media manager bundle lists state not saved to session

### PR DESCRIPTION
**Pull request by pixeljunkie**
| Q             | A
| ------------- | ---
| Branch?       | 6.2.x
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes 
| Fixed issues  | chameleon-system/chameleon-system#4
| License       | MIT

- adds list state save to session
- fixes type conversion of 2 list states

### CODE REVIEW

File: src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php [hunk 1]
```diff
@@ -108,21 +109,42 @@ public function getListState()
...
+        $this->saveStateToSession($state);
+
         return $state;
     }
 
     /**
      * @return null|MediaManagerListState
      */
-    private function getStateFromSession()
+    private function getStateFromSession(): ?MediaManagerListState
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php]: I removed unnecessary docblocks.

File: src/MediaManagerBundle/Resources/public/js/mediaManager.js [hunk 2]
```diff
@@ -628,7 +629,10 @@
             var self = this;
             self.showWaitingAnimation();
             var enableUsageSearch = '0';
-            if ($('.delete-with-usage-search').is(':checked') || '1' === self.state.deleteWithUsageSearch) {
+
+            console.log(self.state.deleteWithUsageSearch);
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/MediaManagerBundle/Resources/public/js/mediaManager.js]: Removed debug output.

File: src/MediaManagerBundle/Resources/public/js/mediaManager.js [hunk 3]
```diff
@@ -34,15 +34,16 @@
...
+            showSubtree: false,
+            deleteWithUsageSearch: false,
             sortColumn: null,
             pickImageMode: false,
             pickImageCallback: '',
             pickImageWithCrop: false
         };
 
     function Plugin(element, state, options) {
+        console.log(state);
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/MediaManagerBundle/Resources/public/js/mediaManager.js]: Removed debug output.

File: src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php [hunk 4]
```diff
@@ -108,21 +109,42 @@ public function getListState()
...
+        return $session->get(self::SESSION_KEY);
+    }
+
+    /**
+     * @param MediaManagerListState $state
+     */
+    private function saveStateToSession(MediaManagerListState $state): void
+    {
+        $session = $this->getSession();
+        $session->set(self::SESSION_KEY, $state);
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php]: now handling possible missing session.

File: src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php [hunk 5]
```diff
@@ -108,21 +109,42 @@ public function getListState()
...
 
     /**
      * @return null|MediaManagerListState
      */
-    private function getStateFromSession()
+    private function getStateFromSession(): ?MediaManagerListState
+    {
+        $session = $this->getSession();
+
+        return $session->get(self::SESSION_KEY);
```
[the remark this answers is not in the document]
Comment (pixeljunkie) [reply src/MediaManagerBundle/Service/MediaManagerListStateFromRequestService.php]: now handling possible missing session.

**Comment by umulmrum** [comment]
Thanks @pixeljunkie

**Pull request merged**